test/FunctionalTests/Microsoft.OData.Core.Tests/Json/JsonWriterBaseTests.cs

```diff
@@ -601,6 +601,121 @@ public void WriteMixedJsonWithLargeValues()
             }
         }
 
+        [Fact]
+        public void CorrectlyStreamsLargeByteArrayToOutput()

```

Comment:
We should also have test cases where the stream and textwriter scopes are created between other JSON values written by the JSON writer. This will help verify that all the data is written to the output stream in the write order. (e.g. `{ "someKey": "someValue", "keyForStreamData": "streamData", "anotherKey": "anotherValue" }